frame/dapps-staking/src/tests.rs

```diff
@@ -1782,6 +1782,198 @@ fn claim_only_payout_is_ok() {
     })
 }
 
+#[test]
+fn claim_only_delegate_once_payout_is_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let developer = 1;
+        let staker = 2;
+        let delegate_account = 3;
+        let contract_id = MockSmartContract::Evm(H160::repeat_byte(0x01));
+
+        // stake some tokens
+        let start_era = DappsStaking::current_era();
+        assert_register(developer, &contract_id);
+        let stake_value = 100;
+        assert_bond_and_stake(staker, &contract_id, stake_value);
+
+        // disable reward restaking
+        advance_to_era(start_era + 1);
+
+        // Set delegated account
+        assert_ok!(DappsStaking::set_delegate_rewards(
+            Origin::signed(staker.clone()),
+            delegate_account,
+            contract_id
+        ));
+
+        let init_balance = Balances::free_balance(delegate_account);
+
+        // ensure it's claimed correctly
+        let reward_total = assert_claim_staker(staker, &contract_id);
```

Comment:
I think there is a problem that i have to solve because i was trying to only allow delegation to account that are FreeBalance. But there is a test "changing_reward_destination_for_empty_ledger_is_not_ok" that don't let it change it if the ledger is empty. So i think the only solution is to create a new RewardDestination. Only allow to delegate to accounts that have this new enum. 

Do you think is a good solution @Dinonard ?